Backend/app/oauth2.py

```
@@ -62,19 +62,32 @@ def __init__(
         super().__init__(flows=flows, scheme_name=scheme_name, auto_error=auto_error)
     
     async def __call__(self, request: Request) -> Optional[str]:
+        authorization = request.headers.get("Authorization")
+        if authorization:
+            scheme, param = get_authorization_scheme_param(authorization)
+            if scheme.lower() == "bearer" and param:
+                try:
+                    verify_token(param)
+                    return param
```

Ruff (0.14.5)

Consider moving this statement to an `else` block

(TRY300)